src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -39,50 +36,42 @@ def acr_import(cmd,
+        else:
             registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry, source_registry)
             if registry:
-                source_registry = registry.id
-        source = ImportSource(resource_id=source_registry, source_image=source_image)
+                source = ImportSource(resource_id=registry.id, source_image=source_image)
+            else:
+                raise CLIError(SOURCE_REGISTRY_NOT_FOUND)
     else:
-        source_registry_login_server = source_image[:slash]
-        source_image = source_image[slash + 1:]
-        if not source_image or not source_registry_login_server:
-            raise CLIError(INVALID_SOURCE_IMAGE)
-        registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry_login_server)
-        if registry:
-            if source_registry and \
-               source_registry.lower() != registry.id.lower() and \
-               source_registry.lower() != registry.name.lower() and \
-               source_registry.lower() != registry.login_server.lower():
-                raise CLIError(REGISTRY_MISMATCH)
-            source = ImportSource(resource_id=registry.id,
-                                  source_image=source_image)
-        else:
+        slash = source_image.find('/')
+        if slash > 0:
+            registry_uri = source_image[:slash]
+            source_image = source_image[slash + 1:]
             if source_registry_password:
                 ImportSourceCredentials = cmd.get_models('ImportSourceCredentials')
                 if source_registry_username:
-                    source = ImportSource(registry_uri=source_registry_login_server,
+                    source = ImportSource(registry_uri=registry_uri,
                                           source_image=source_image,
                                           credentials=ImportSourceCredentials(password=source_registry_password,
                                                                               username=source_registry_username))
                 else:
-                    source = ImportSource(registry_uri=source_registry_login_server,
+                    source = ImportSource(registry_uri=registry_uri,
                                           source_image=source_image,
                                           credentials=ImportSourceCredentials(password=source_registry_password))
             else:
-                source = ImportSource(registry_uri=source_registry_login_server, source_image=source_image)
+                registry = get_registry_from_name_or_login_server(cmd.cli_ctx, registry_uri)
+                if registry:
+                    source = ImportSource(resource_id=registry.id, source_image=source_image)
+                else:
+                    source = ImportSource(registry_uri=registry_uri, source_image=source_image)
+        else:
+            raise CLIError(INVALID_SOURCE_IMAGE)
 
     if not target_tags and not repository:
```

Comment:
Is it a valid scenario if both `target_tags` and `repository` are specified?

Reply:
It is valid. Additionally, if you mean two source images are present in the command, one being in the fully qualified source and the other being the repository, both are imported regardless of the images exist or not. Target_tags and repository by user input are for naming the target image in the target registry only.

Reply:
You mean if I specify `--source image1 --source image2`? I thought only the last one will be imported?

Reply:
If I do `--source image1 --repository repository2`, image1 and all the images in repository2 are imported.

Reply:
This question is answered. Just `--repository` is target, not source.

Reply:
I mean if the command is: `az acr import -n targetRegistry --source sourceRegistry.azurecr.io/repo1:tag1 -t targetRepo:targetTag --repository repo2`, then two things are imported, one being repo1:tag1 and it will be tagged as targetRepo:targetTag in the target registry, the other being repo2 and there will be no retagging.